components/server/ee/src/workspace/gitpod-server-impl.ts

```diff
@@ -2174,9 +2174,13 @@ export class GitpodServerEEImpl extends GitpodServerImpl {
                 throw new Error(`No Stripe customer profile for '${attributionId}'`);
             }
 
+            //TODO: feature flag with below
```

Comment:
Here, you don't have the Stripe Customer ID, how are you going to retreive it?
And if you're gonna retrieve it from the usage component, why do we need to provide back into the `CreateSubscription` call?
Can we instead remove it from `CreateSubscription` and look it up from the Attribution ID?

Reply:
`customerId` is available in the line above[[1](https://github.com/gitpod-io/gitpod/blob/0c7a0a6834b2bf3d7156ba87330b16cce1cf1c9c/components/server/ee/src/workspace/gitpod-server-impl.ts#L2172)].
I tried that but had the problem with the race condition that I mentioned. If it's already available, isn't it easier to use that instead?

Reply:
That race condition shouldn't really occur. In any case, doing the lookup above actually goes to the usage component anyway, so it's the same flow and would manifest in both cases AFAIK.

Reply:
I'm happy to pair on this part since I ran into the issue multiple times (hence the change), so maybe I did it incorrectly 🙏🏽